app.py

```diff
@@ -309,6 +310,75 @@ def uploaded_file(filename):
     """
     return send_from_directory(app.config['UPLOAD_FOLDER'],filename)
 
+@app.route('/metric-csv', methods=['GET'])
+def get_metrics_as_csv():
+    """
+    **GET Request Parameters**
+    > `metric`
+    > `start_timestamp`
+    > `end_timestamp`
+    """
+    try:
+        # get request parameters
+        metric = request.args.get("metric")
+        start_timestamp = request.args.get("start_timestamp")
+        end_timestamp = request.args.get("end_timestamp")
+
+        #throw error if any required parameters are None
+        if None in [metric, start_timestamp, end_timestamp]:
+            raise Error("NULL_VALUE")
+
+        # generate all function queries
+        fullness_query = models.BinFullness.query.filter(and_(start_timestamp <= models.BinFullness.datetimestamp, end_timestamp >= models.BinFullness.datetimestamp)).all
+        weight_query = models.BinWeight.query.filter(and_(start_timestamp <= models.BinWeight.datetimestamp, end_timestamp >= models.BinWeight.datetimestamp)).all
+        usage_query = models.BinUsage.query.filter(and_(start_timestamp <= models.BinUsage.datetimestamp, end_timestamp >= models.BinUsage.datetimestamp)).all
+
+        # generate dict of functions for querying specific data metric
+        data_query_func = {"fullness":fullness_query, "weight":weight_query, "usage":usage_query}
+
+        # check if parameters are valid
+        if metric not in data_query_func.keys():
+            raise Error("INVALID_PARAM")
+
+        #throw error if start_timestamp > end_timestamp
+        if start_timestamp > end_timestamp:
+            raise Error("TIMESTAMP_ISSUE")
+
+        # call query function
+        query_data = data_query_func[metric]()
+
+        # iterate through each row of data
+        obj_dict = None
+        for row in query_data:
+            # convert fullness data into dictionary form such as {key1:str1, key2:str2}
+            row_dict_form = dict(row)
+
+            # assign obj_dict to appropriate keys based on model schema
+            if not obj_dict:
+                headers = row_dict_form.keys()
+                # create a dictionary of empty lists such as {key1:[], key2:[]}
+                obj_dict = dict(zip(headers, [[] for i in range(len(headers))]))
+
+            # append data to the obj_dict based on keys
+            for k in obj_dict.keys():
+                obj_dict[k].append(row_dict_form[k])
+
+        # create the pandas dataframe
+        df = pandas.DataFrame(obj_dict)
+        resp = make_response(df.to_csv(index=False))
+        file_name = "{}_{}_{}.csv".format(metric,start_timestamp,end_timestamp)
```

Comment:
If I remember correctly, the timestamp has a `:` in it, and that character is not allowed for the filename on Windows. Make sure that the filename has none of these characters `NUL, \, /, :, *, ", <, >, |`